vslib/inc/MACsecFilterStateGuard.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "MACsecFilter.h"
+
+namespace saivs
+{
+    class MACsecFilterStateGuard
+    {
+    public:
+
+        MACsecFilterStateGuard(
+            _Inout_ volatile MACsecFilter::MACsecFilterState &guarded_state,
```

Comment:
seems like a overkill here, did you hit scenario that this was optimized in wrong way by compiler that you need volatile here?

Reply:
Agree, it looks like overkill. I will remove the qualifier.